proc_macros/src/lib.rs

```diff
@@ -1,13 +1,58 @@
 use std::collections::HashMap;
 
+use inflector::Inflector;
 use proc_macro::TokenStream;
-use quote::{quote, ToTokens};
+use quote::{format_ident, quote, ToTokens};
 use syn::{
     braced, bracketed,
-    parse::{Parse, ParseStream},
-    parse_macro_input, Expr, ExprArray, ExprPath, Ident, Token,
+    parse::{Parse, ParseStream, Parser},
+    parse_macro_input, parse_quote,
+    punctuated::Punctuated,
+    token,
+    visit_mut::{self, VisitMut},
+    Expr, ExprArray, ExprClosure, ExprField, ExprMacro, ExprPath, Ident, Member, Token, Type,
 };
 
+struct ArrowSeparatedKeyValuePair<TKey = Ident, TValue = Expr> {
```

Comment:
Started to abstract out the `something => value` proc-macro syntax "building block"